### DISCUSSION TITLE
How to add sticky header to table

### DISCUSSION THREAD

**Discussion by Ivcast20** [rappasoft/laravel-livewire-tables#1100] [Q&A]
Pleaseee I want to Know if there's a way to make a Sticky header to a datatable.

**Comment by lrljoe** [comment]
What do you mean by Sticky Header?

**Reply by Ivcast20** [reply]
Well, When I got a table that has a lot of information, when I scroll down the header disapears. I want to know if there's a way to keep showing the header when I scroll down.

**Reply by lrljoe** [reply]
Are you using tailwind or bootstrap?

**Reply by Ivcast20** [reply]
I'm using bootstrap

**Reply by lrljoe** [reply]
I'll have to take a look over next few days for BS.

In theory you just need to pass "sticky top-0" to the thead and its tr element(s).  How bootstrap will behave is not my specialism so I'll need to take a look.

**Reply by lrljoe** [reply]
Yeah, you'll need to pass the correct classes to make a table header sticky using:
setTableAttributes
setTheadAttributes
and
setThAttributes

I'll put a quick example together, but you'll need to tweak it to get it looking exactly how you'd like it.

**Reply by Ivcast20** [reply]
Please if you can help me with an example I will apreciate it.

**Reply by lrljoe** [reply]
![image](https://user-images.githubusercontent.com/104938042/226084697-677df486-8b48-4adc-aac7-97fcc5b3a0c9.png)

Then in your configure do something like the following:
```php
$this->setTableWrapperAttributes([
      'default' => false,
      'class' => '',
    ])
  ->setTableAttributes([
      'class' => 'w-full border',
  ])
  ->setTheadAttributes([
      'class' => 'sticky-top',
  ]);
```

That will set your table to be full-width, and for the Thead to be sticky.  You'll need to style it appropriately, as otherwise you'll notice that it won't have a decent background.  Not sure how it'll work responsively, but it gives you a good starting point.

Tailwind is a much easier learning curve by the way!
